src/main/java/com/kodilla/ecommercee/service/UserService.java

```diff
@@ -0,0 +1,63 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.exception.UserNotFoundException;
+import com.kodilla.ecommercee.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Random;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public List<User> getUsers() {
+        return userRepository.findAll();
+    }
+
+    public User getUser(Long id) {
+        return userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+    }
+
+    public void deleteUser(Long id) throws UserNotFoundException {
+        if(userRepository.findById(id).isPresent()) {
+            userRepository.deleteById(id);
+        } else {
+            throw new UserNotFoundException();
+        }
+    }
+
+    public void updateUser(Long id, User user) throws UserNotFoundException {
```

Comment:
```suggestion
    public void updateUser(Long id, User user)  {
```

Reply:
I would make a seperate method of checking if user exists and re-use it, here and in deleteUser also  throw an exception there so you would get rid of 2 - if statements

Reply:
In general, it's a good and professional approach, but in this case I would gain a maximum of one or two lines of code, while losing minimally on speed (yes - I know these are negligible differences), but also some readability.
I don't think it's worth it with so little code - if it was about another method, i would actually do it.